src/workerd/jsg/filesystem.c++

```diff
@@ -0,0 +1,537 @@
+#include "filesystem.h"
+
+#include "buffersource.h"
+
+namespace workerd::jsg::fs {
+
+namespace {
+constexpr size_t kMaxReadLength = 32 * 1024;
+
+void validatePathWithNoSeparators(kj::StringPtr path) {
```

Comment:
Can you add a comment to functions similar to this? what is the intended result etc. Whether or not it will throw.